app/models/payment_providers/adyen_provider.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module PaymentProviders
+  class AdyenProvider < BaseProvider
+    validates :api_key, :merchant_account, presence: true
+
+    def environment
+      if Rails.env.production? && live_prefix.present?
+        :live
+      else
+        :test
+      end
+    end
+
+    def api_key=(value)
+      push_to_secrets(key: 'api_key', value:)
+    end
+
+    def api_key
+      get_from_secrets('api_key')
+    end
+
+    def merchant_account=(value)
+      push_to_secrets(key: 'merchant_account', value:)
+    end
+
+    def merchant_account
+      get_from_secrets('merchant_account')
+    end
+
+    def live_prefix=(value)
+      push_to_secrets(key: 'live_prefix', value:)
```

Comment:
I don't know if this one is a sensitive , maybe using `push_to_settings` and `get_from_settings` is enough. What do you think?

Maybe it is similar for the `merchant_account` attribute?

Reply:
You're right, I agree, changing it that way - both `live_prefix` and `merchant_account`